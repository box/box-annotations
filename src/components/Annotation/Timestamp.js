// @flow
import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import { ReadableTime } from 'box-react-ui/lib/components/time';
import Tooltip from 'box-react-ui/lib/components/tooltip';

import messages from './messages';
import './Timestamp.scss';

const ONE_HOUR_MS = 3600000; // 60 * 60 * 1000

type Props = {
    time: string
};

const Timestamp = ({ time }: Props) => {
    const createdAtTimestamp = new Date(time).getTime();

    return (
        <Tooltip
            text={<FormattedMessage {...messages.annotationPostedFullDateTime} values={{ time: createdAtTimestamp }} />}
        >
            <time className='ba-annotation-created-at'>
                <ReadableTime timestamp={createdAtTimestamp} relativeThreshold={ONE_HOUR_MS} />
            </time>
        </Tooltip>
    );
};

export default Timestamp;
