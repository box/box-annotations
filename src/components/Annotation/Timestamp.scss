@import '../../variables.scss';

.ba .ba-annotation-created-at {
    color: $nines;
    cursor: default;
    font-size: smaller;
}

.ba-mobile-annotation-dialog .ba-annotation-created-at {
    font-size: 14px;
}
