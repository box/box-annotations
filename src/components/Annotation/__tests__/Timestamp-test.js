import * as React from 'react';
import { shallow } from 'enzyme';

import Timestamp from '../Timestamp';

const TIME_STRING_SEPT_27_2017 = '2017-09-27T10:40:41-07:00';

describe('components/Annotation/Timestamp', () => {
    test('should correctly render annotation', () => {
        const unixTime = new Date(TIME_STRING_SEPT_27_2017).getTime();

        const wrapper = shallow(<Timestamp time={TIME_STRING_SEPT_27_2017} />).dive();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('ReadableTime').prop('timestamp')).toEqual(unixTime);
    });
});
