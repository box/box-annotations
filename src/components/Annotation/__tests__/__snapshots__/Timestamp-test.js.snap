// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Annotation/Timestamp should correctly render annotation 1`] = `
<TetherComponent
  attachment="bottom center"
  bodyElement={<body />}
  classPrefix="tooltip"
  constraints={
    Array [
      Object {
        "attachment": "together",
        "to": "window",
      },
    ]
  }
  enabled={false}
  renderElementTag="div"
  renderElementTo={null}
  targetAttachment="top center"
>
  <time
    className="ba-annotation-created-at"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex="0"
  >
    <ReadableTime
      relativeThreshold={3600000}
      timestamp={1506534041000}
    />
  </time>
</TetherComponent>
`;
